### PR TITLE
[IOPAY-146] error from nodo pagoPa responses + mixpanel events

### PR DESCRIPTION
**Pull request by pasqualespica**
This PR add the capability to manage the following semantic errors, related to verify and activation phase with `pagopa-proxy`: 
- PAYMENT_DUPLICATED
- INVALID_AMOUNT
- PAYMENT_ONGOING
- PAYMENT_EXPIRED
- PAYMENT_UNAVAILABLE
- PAYMENT_UNKNOWN

and mixpanel events

#### List of Changes

- Update `response.ts` to manage error internal response 
- add custom modal call on error response from `pagopa-proxy`
- add `mixpanel.track` event in to verify, activation and get payment with `pagopa-proxy`

#### Motivation and Context

To show a specific errore message to user as "Avviso già pagato"
See here to details [IOPAY-146](https://pagopa.atlassian.net/browse/IOPAY-146)

Add [mixpanel](https://mixpanel.com/) event , see [IOPAY-170](https://pagopa.atlassian.net/browse/IOPAY-170)

#### How Has This Been Tested?

_Manual test_
1. Run local `io-functions-pay-portal` typing `yarn start`
2. Run local `io-mock` typing 
> with custom and modify `io-pagopa-node-mock` ( using `docker-compose.override.yml` to do it )
3. Send a **GET** http://localhost:7071/api/v1/payment-requests/77234567891010001234567890123

To verify mixpanel event add `IO_PAY_PORTAL_ENV=develop` before starting frontend as follow :
`IO_PAY_PORTAL_ENV=develop yarn dev:proxy` 
in this way al `mixpanel.track` call are mock on `console.log` and could be inspect by _devtools_

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: io-functions-pay-portal/utils/responses.ts [hunk 1]
```diff
@@ -63,6 +65,12 @@ export const toErrorServerResponse = <S extends number, T>(
       return ResponseErrorNotFound("Not found", "Resource not found");
     case 429:
       return ResponseErrorTooManyRequests("Too many requests");
+    case 500:
+      return ResponseErrorInternal(
+        PaymentProblemJson.decode(response.value).getOrElse({
+          detail: PaymentFaultEnum.PAYMENT_UNKNOWN
+        }).detail || "PaymentProblemJson on decode"
```
Comment (infantesimone) [inline io-functions-pay-portal/utils/responses.ts:72]: This is a default error, for example when the _proxy-node_ returns 500 without a body, maybe I would change to "Generic Error".

File: io-pay-portal-fe/src/helper.ts [hunk 2]
```diff
@@ -28,20 +56,48 @@ export const getPaymentInfoTask = (
...
+      return fromLeft(err);
+    },
     (errorOrResponse) =>
       errorOrResponse.fold(
         () => fromLeft("Errore recupero pagamento"),
-        (responseType) =>
-          responseType.status !== 200
-            ? fromLeft(`Errore recupero pagamento : ${responseType.status}`)
-            : taskEither.of(responseType.value)
+        (responseType) => {
```
Comment (infantesimone) [inline io-pay-portal-fe/src/helper.ts:83]: maybe you can simplify it with _fromPredicate_  or:

```
     const EVENT_ID : string = responseType.status === 200 ? 
          PAYMENT_VERIFY_SUCCESS.value : 
          PAYMENT_VERIFY_RESP_ERR.value ;
     mixpanel.track(EVENT_ID, { EVENT_ID });
```

**Review by infantesimone — APPROVED** [review]
well done 👍

**Comment by infantesimone** [comment]
There are some conflict, to rebase and resolve conflict @pasqualespica

**Comment by pasqualespica** [comment]
> There are some conflicts, to rebase and resolve conflict @pasqualespica

tnx , done

**Pull request merged**